internal/fields/validate.go

```diff
@@ -190,7 +190,19 @@ func parseElementValue(key string, definition fieldDefinition, val interface{})
 	var valid bool
 	switch definition.Type {
 	case "date", "ip", "constant_keyword", "keyword", "text":
-		_, valid = val.(string)
+		var valStr string
+		valStr, valid = val.(string)
+		if !valid || definition.Pattern == "" {
+			break
+		}
+
+		valid, err := regexp.MatchString(definition.Pattern, valStr)
+		if err != nil {
+			return errors.Wrap(err, "invalid pattern")
+		}
+		if !valid {
+			return fmt.Errorf("field \"%s\" has invalid format, pattern: %s, actual: %s", key, definition.Pattern, valStr)
```

Comment:
I think we need to start making our error messages, especially validation ones, more user-friendly (related: https://github.com/elastic/package-spec/issues/75). So I would suggest changing this to something like:

```suggestion
			return fmt.Errorf("field \"%s\"''s value, %s, does not match the expected pattern: %s", key, valStr, definition.Pattern)
```

Reply:
Good idea, I updated the PR.